### PR TITLE
Include environment details in CI output

### LINKED ISSUE #81: Add script for printing environment details in CI
Include output of `nvidia-smi` and conda environment details

### PR DESCRIPTION
**Pull request by wphicks**
Include the output of `nvidia-smi` and `conda list` in CI test output
Optionally include this information in local test runs via the `SHOW_ENV` flag

Resolve #81

### CODE REVIEW

**Comment by wphicks** [comment]
@dantegd Could I trouble you for review on this one? Anything else you'd like to see included in this output?

File: qa/run_tests.sh [hunk 1]
```diff
@@ -18,6 +22,20 @@ fi
...
+then
+  echo '-------------------------------------------------------'
+  echo '----------------- Environment details -----------------'
+  echo '-------------------------------------------------------'
+  echo ''
+  echo '---------------------- nvidia-smi ---------------------'
+  nvidia-smi
+  echo ''
+  echo '------------------------ conda ------------------------'
+  conda list -q
```
Comment (dantegd) [inline qa/run_tests.sh:35]:
```suggestion
  conda info
  conda list -q
```

File: qa/run_tests.sh [hunk 2]
```diff
@@ -18,6 +22,20 @@ fi
...
 
+if [ $SHOW_ENV -eq 1 ]
+then
+  echo '-------------------------------------------------------'
+  echo '----------------- Environment details -----------------'
+  echo '-------------------------------------------------------'
+  echo ''
+  echo '---------------------- nvidia-smi ---------------------'
+  nvidia-smi
+  echo ''
```
Comment (dantegd) [inline qa/run_tests.sh:33]:
```suggestion
  echo ''
  echo '---------------------- environment -------------------'
  env
  echo ''
```

**Review by dantegd — CHANGES_REQUESTED** [review]
Just a couple of suggestions that can potentially help for making the logs more useful

File: qa/run_tests.sh (hunk 2 above)
Comment (wphicks) [reply qa/run_tests.sh:33]: My only hesitation on this one is that if people submit an issue with output from a run with `SHOW_ENV=1` enabled, they may not want to expose all their environment variables. Maybe we can enable this only for `[ $LOCAL == 0 ] && [ SHOW_ENV ==1 ]`?

**Comment by wphicks** [comment]
CI failure is unrelated; I'll rerun it as soon as I fix what I broke...

File: qa/run_tests.sh (hunk 2 above)
Comment (dantegd) [reply qa/run_tests.sh:33]: that sounds like a good idea indeed

**Comment by wphicks** [comment]
CI is fixed, and the additional output has been added. I think this is ready for re-review.

**Review by dantegd — APPROVED** [review]

**Pull request merged**